packages/subiquity_client/lib/src/client.dart

```diff
@@ -208,6 +214,21 @@ class SubiquityClient {
     await _receive("setIdentity(${jsonEncode(identity.toJson())})", response);
   }
 
+  Future<UsernameValidation> validateUsername(String username) async {
+    final request = await _openUrl(
+      'GET',
+      Uri.http(
+        'localhost',
+        'identity/validate_username',
+        {'username': '"$username"'},
+      ),
+    );
+    final response = await request.close();
+
+    final respStr = await _receive("identity/validate_username()", response);
+    return UVString.fromString(respStr.removePrefix('"').removeSuffix('"'));
```

Comment:
you can replace `UVString.fromString()` with `UsernameValidation.values.byName()`

https://api.dart.dev/stable/2.17.1/dart-core/EnumByName.html